client/client.go

```diff
@@ -144,6 +147,24 @@ func New(cfg Config) (Client, error) {
 	return c, nil
 }
 
+// KeepSync keeps synchronizing with etcd cluster about its member list,
+// so client could update its endpoints and catch up the change of the cluster.
+func KeepSync(ctx context.Context, c Client) error {
```

Comment:
I don't want to name it `AutoSync` because it doesn't happen automatically or in background, users call it explictly and it will block as long as user wants it to be sync periodically.